src/rgw/rgw_json_enc.cc

```diff
@@ -107,6 +127,9 @@ void RGWObjManifest::dump(Formatter *f) const
   ::encode_json("rules", rules, f);
   ::encode_json("tail_instance", tail_instance, f);
   ::encode_json("tail_placement", tail_placement, f);
+
+  f->dump_object("begin_iter", begin_iter);
```

Comment:
should be able to use encode_json() instead, probably better using it instead of all the direct formatter->dump() calls

Reply:
In my previous pr(#18959 ), @tchaikov suggest to use formatter's method.  It confused me which should be used. 
![image](https://user-images.githubusercontent.com/2369293/33983726-299268e4-e0f0-11e7-8306-0650b4d85c5c.png)

Reply:
@yaozongyou well, it depends on if we are dumping a JSON or not. but in this case, i think `RGWObjManifest::dump()` is a method accepting a general `Formatter`, why shall we use `encode_json()`?

Reply:
@tchaikov encode_json's last parameter is a general ceph::Formatter，it has the same capability with formatter->dump. and in this source code file, as the original GWObjManifest::dump using encode_json, we'd better keeping consistent with it.

Reply:
> it has the same capability with formatter->dump.

also please note that the same applies to to `encode_json()`. and the latter is a wrapper of the `formatter->dump()`, extra cpu cycles wasted if not inlined.

> as the original GWObjManifest::dump using encode_json, we'd better keeping consistent with it.

`Foo::dump()` is a general interface used by `Formatter` not just JSON dump methods, it's consistent to follow this convention. but i understand, the to use `encode_json()` is also consistent with the rest part of this method or this source file. so no strong feelings either way. the reason i am chiming in here is merely that i am @'ed.

Reply:
as @tchaikov has no favor for either way and @mattbenjamin has approved it, @cooboos let's keep it as it is.

Reply:
@yaozongyou ok， thank you !